airflow/providers/databricks/hooks/databricks_base.py

```diff
@@ -0,0 +1,373 @@
+                AZURE_METADATA_SERVICE_INSTANCE_URL,
+                params={"api-version": "2021-02-01"},
+                headers={"Metadata": "true"},
+                timeout=2,
+            ).json()
+            if 'compute' not in jsn or 'azEnvironment' not in jsn['compute']:
+                raise AirflowException(
+                    f"Was able to fetch some metadata, but it doesn't look like Azure Metadata: {jsn}"
+                )
+        except (requests_exceptions.RequestException, ValueError) as e:
+            raise AirflowException(f"Can't reach Azure Metadata Service: {e}")
+
+    def _get_token(self, raise_error: bool = False) -> Optional[str]:
+        if 'token' in self.databricks_conn.extra_dejson:
+            self.log.info(
+                'Using token auth. For security reasons, please set token in Password field instead of extra'
+            )
+            return self.databricks_conn.extra_dejson['token']
+        elif not self.databricks_conn.login and self.databricks_conn.password:
+            self.log.info('Using token auth.')
+            return self.databricks_conn.password
+        elif 'azure_tenant_id' in self.databricks_conn.extra_dejson:
+            if self.databricks_conn.login == "" or self.databricks_conn.password == "":
+                raise AirflowException("Azure SPN credentials aren't provided")
+            self.log.info('Using AAD Token for SPN.')
+            return self._get_aad_token(DEFAULT_DATABRICKS_SCOPE)
+        elif self.databricks_conn.extra_dejson.get('use_azure_managed_identity', False):
+            self.log.info('Using AAD Token for managed identity.')
+            self._check_azure_metadata_service()
+            return self._get_aad_token(DEFAULT_DATABRICKS_SCOPE)
+        elif raise_error:
+            raise AirflowException('Token authentication isn\'t configured')
+
+        return None
+
+    @staticmethod
+    def _retryable_error(exception) -> bool:
+        return (
+            isinstance(exception, (requests_exceptions.ConnectionError, requests_exceptions.Timeout))
+            or exception.response is not None
+            and exception.response.status_code >= 500
+        )
+
+    def _log_request_error(self, attempt_num: int, error: str) -> None:
+        self.log.error('Attempt %s API Request to Databricks failed with reason: %s', attempt_num, error)
+
+    def _do_api_call(self, endpoint_info: Tuple[str, str], json: Optional[Dict[str, Any]] = None):
+        """
+        Utility function to perform an API call with retries
+
+        :param endpoint_info: Tuple of method and endpoint
+        :param json: Parameters for this API call.
+        :return: If the api call returns a OK status code,
+            this function returns the response in JSON. Otherwise,
+            we throw an AirflowException.
+        :rtype: dict
+        """
+        method, endpoint = endpoint_info
+
+        # TODO: get rid of explicit 'api/' in the endpoint specification
```

Comment:
yes, because this function is always call API, then it makes no sense to have `api/` prefix for all declared endpoints. Then it's easier to work. But it will be bigger rework, as a separate PR